web/src/pages/billing/BillingCostByTime.tsx

```diff
@@ -335,16 +335,69 @@ const BillingCostByTime: React.FunctionComponent = () => {
         }
     }, [start, end, groupBy, selectedData])
 
+    const exportOptions = [
+        { key: 'csv', text: 'Export to CSV', value: 'csv' },
+        { key: 'tsv', text: 'Export to TSV', value: 'tsv' },
+    ]
+
+    const exportToFile = (format: 'csv' | 'tsv') => {
+        // Create header: Date plus group/column names
+        const headerData = ['Date', ...groups]
+
+        // Build rows: for each object in `data` (each date)
+        const rowData = data.map((row) => {
+            // Each `row` is { date, values }
+            const dateStr =
+                row.date instanceof Date ? row.date.toISOString().slice(0, 10) : String(row.date)
```

Comment:
similar comment re. the date here, row.date should always be a date so no need to check for it.